crit/cmd/cli.go

```diff
@@ -71,11 +72,17 @@ var encodeCmd = &cobra.Command{
 	Short: "Convert JSON to binary file",
 	Long:  "Convert the input JSON to a CRIU image file.",
 	Run: func(cmd *cobra.Command, args []string) {
-		c = crit.New(inputFilePath, outputFilePath, inputDirPath, exploreType, pretty, noPayload)
-		_, err := c.Dump()
+		c = crit.New(inputFilePath, outputFilePath,
+			inputDirPath, exploreType, pretty, noPayload)
+		// Convert JSON to Go struct
+		img, err := c.Parse()
 		if err != nil {
```

Comment:
This error check could be on a single line:
```go
if img, err := c.Parse(); err != nil {
}
```

Reply:
Making it a one-liner reduces the scope of `img` to the `if` block. I'm using it in line 83 for `c.Dump(img)`